mantidproject/mantid
# Memory widget on the workbench main-window

**Description of work.**


Added memory widget to the workbench main-window, to display the total memory usage

Report to: ISIS/Diego Alba-Venero and Refl

**To test:**


-> Pull and Checkout  branch "30757_memory_bar"

-> **On Windows**
    -> open "Run" app and type "%APPDATA%" and Enter
    -> In the revealed folder (.../AppData/Roaming/), go to "mantidproject" folder
    -> In the "mantidproject" folder, delete "mantidworkbench.ini" file

-> **On Mac**
    -> Remove "~/.config/mantidproject/mantidworkbench.ini" file

-> Open the workbench application, and memory widget should appear at the
     top right corner displaying memory usage info.

-> The memory_bar colour fill should be light-green, if the memory usage is sub-critical (<90%)
     or the colour fill should be red if the memory usage is critical (>= 90%)

**On Windows**
![on Windows](https://user-images.githubusercontent.com/52707387/112305813-08c4d600-8c97-11eb-971e-bf1d8115e704.png)

**On Mac** (Thanks to @DanielMurphy22)
![On Mac](https://user-images.githubusercontent.com/52707387/112306268-8ab4ff00-8c97-11eb-9a03-ca1c25b8d670.png)


Resolves #30757






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
